src/_pytest/fixtures.py

```diff
@@ -256,21 +262,25 @@ def get_parametrized_fixture_keys(item: nodes.Item, scope: Scope) -> Iterator[_K
         # cs.indices.items() is random order of argnames.  Need to
         # sort this so that different calls to
         # get_parametrized_fixture_keys will be deterministic.
-        for argname, param_index in sorted(cs.indices.items()):
+        for argname in sorted(cs.indices):
             if cs._arg2scope[argname] != scope:
                 continue
+
+            item_cls = None
             if scope is Scope.Session:
-                key: _Key = (argname, param_index)
+                scoped_item_path = None
             elif scope is Scope.Package:
-                key = (argname, param_index, item.path)
+                scoped_item_path = item.path.parent
```

Comment:
There's a semantic conflict here, the `item.path.parent` changed to `item.path` in main (a21fb87a90974189c1b8b26189959507189bb3a1). However, I just realized that that change was wrong, and that `item.path.parent` is somewhat wrong as well.

Anyway, let's keep it `scoped_item_path = item.path` here, because changing it is not the purpose of this PR, and I will send a PR to fix it after (the fix will actually be easier with the new `FixtureArgKey`).

Reply:
Why is `item.path.parent` wrong? Because packages could contain subdirectories? How is this?
```python
elif scope is Scope.Package:
    if isinstance(item.parent, Package):
        scoped_item_path = item.parent.path
    elif isinstance(item.parent.parent, Package):
        scoped_item_path = item.parent.parent.path # If item is a class method
```

Reply:
As you said, this test fails currently:
```python
def test_reordering_with_package_scoped_parametrization_of_test_in_subdirectory(pytester: Pytester) -> None:
    pytester.makepyfile(
        __init__= "",
        test_a= textwrap.dedent(
        """\
        import pytest

        @pytest.mark.parametrize("arg", [0], scope="package")
        def test_1(arg):
            pass
        
        def test_2():
            pass
        """,
        ),
    )
    subdir = pytester.mkdir("subdirectory")
    subdir.joinpath("test_b.py").write_text(
        textwrap.dedent(
            """\
            import pytest

            @pytest.mark.parametrize("arg", [0], scope="package")
            def test_3(arg):
                pass
            """,
        ),
        encoding="utf-8",
    )
    result = pytester.runpytest("--collect-only")
    result.stdout.fnmatch_lines(
        [
        "*test_1*",
        "*test_3*",
        "*test_2*",
        ]
    )
```

Reply:
And this also fails for class branch:
```python
def test_reordering_with_class_scoped_parametrization_of_classless_test(pytester: Pytester) -> None:
    pytester.makepyfile(
        """
        import pytest
        
        @pytest.mark.parametrize("arg", [0], scope="class")
        def test_1(arg):
            pass
            
        def test_2():
            pass
        
        @pytest.mark.parametrize("arg", [0], scope="class")
        def test_3(arg):
            pass
    """
    )
    result = pytester.runpytest("--collect-only")
    result.stdout.fnmatch_lines(
        [
        "*test_1*",
        "*test_2*",
        "*test_3*",
        ]
    )
```

Reply:
If we set `scoped_item_path` to `None` when the package-scoped test is package-less and do not skip the key, then the test would have common key with all the package-scoped tests that have the same parameter, so they will be gathered together in the reordering. Is this the desired behavior? If we skip the key, the test gets isolated up until it's no longer package-less.
This is also the case for class branch. Do we want class-less tests using a parameter to be gathered altogether in reordering or be isolated?

Reply:
If we skip the key, it might seem more intuitive to the user. He/She says: I've set scope x for the test. It is not located in a x. So the scope would be ineffective.